src/extension.ts

```diff
@@ -196,9 +200,49 @@ function registerDockerClients(): void {
         if (e.affectsConfiguration('docker.composeCommand')) {
             composeClient.reconfigure();
         }
+
+        if (e.affectsConfiguration('docker.contexts.showInStatusBar')) {
+            // Don't wait
+            void registerContextStatusBarItems(ext.context);
+        }
     });
 }
 
+async function registerContextStatusBarItems({ subscriptions }: vscode.ExtensionContext) {
+    const currentContext = await ext.runtimeManager.contextManager.getCurrentContext();
+    const showInStatusBar = vscode.workspace.getConfiguration('docker').get('contexts.showInStatusBar', false);
+
+    // if it's undefined, it means there is no context set, so we don't need to show the status bar item
+    // if user don't want context to clutter up status bar, we don't need to show
+    if (!currentContext || !showInStatusBar) {
```

Comment:
I'd set this up so that we check `showInStatusBar` first, and don't check `currentContext` if `showInStatusBar` is false. The context lookup could potentially be slow, and will result in communicating with Docker before it's necessary. Something like this:

```typescript
const config = vscode.workspace.getConfiguration('docker');
if (!config.get('contexts.showInStatusBar', false)) {
    dockerContextStatusBarItem?.dispose();
    return;
}

const currentContext = await ext.runtimeManager.contextManager.getCurrentContext();
if (!currentContext) {
    dockerContextStatusBarItem?.dispose();
    return;
}
```

or

```typescript
const config = vscode.workspace.getConfiguration('docker');
let currentContext: ListContextItem | undefined;

if (!config.get('contexts.showInStatusBar', false) ||
    !(currentContext = await ext.runtimeManager.contextManager.getCurrentContext())) { // Intentional assignment and boolean check
    dockerContextStatusBarItem?.dispose();
    return;
}
```

Reply:
that's a great point, context checking does take around 1~ second and this would avoid unnecessary slower runtime